File: spoj/42-ADDREV/solution_42.cpp
#include <iostream>

using namespace std;

int revNum(int);

int main()
{
    int t;
    cin >> t;
    for (int i = 0; i < t; i++)
    {
        int x, y;
        cin >> x >> y;
        int revX = revNum(x);
        int revY = revNum(y);
        cout << revNum(revX + revY) << "\n";
    }
}

int revNum(int num)
{

    int rev = 0;

    while (num != 0)
    {
        int digit = num % 10;
        rev = (rev * 10) + digit;
        num /= 10;
    }

    return rev;
}

File: spoj/42-ADDREV/solution_42.java
import java.util.Scanner;

public class solution_42 {
    private static Scanner sc;

    public static void main(String[] args) {
        sc = new Scanner(System.in);

        int n = sc.nextInt();

        for (int i = 0; i < n; i++) {
            int x = sc.nextInt();
            int y = sc.nextInt();
            int revX = revNumber(x);
            int revY = revNumber(y);
            System.out.println(revNumber(revX + revY));
        }
    }

    public static int revNumber(int num) {
        int rev = 0;
        while (num != 0) {
            int digit = num % 10;
            rev = rev * 10 + digit;
            num /= 10;
        }

        return rev;
    }
}